src/checkers.cpp

```diff
@@ -51,5 +51,36 @@ namespace ada::checkers {
     }
     return accumulator;
   }
+
+  ada_really_inline constexpr bool check_domain(std::string_view input) noexcept {
+    if(input.size() > 255) {
+      return false;
+    }
+
+    const char* start = input.data();
+    const char* end = start + input.size();
+
+    int dot_count = 0;
+    while (start < end) {
+        // Find the next dot in the domain
+        const char* dot = std::find(start, end, '.');
```

Comment:
I believe `size_t dot_location = input.find('.', left_padding);` would reduce the number of instructions.

Reply:
We have the instrumentation to count the instructions, so we can explore the design space as needed.